Dockerfile

```diff
@@ -1,41 +1,39 @@
-FROM ubuntu:22.04
-USER root
 ARG DEBIAN_FRONTEND=noninteractive
-
 # update and upgrade packages
 RUN apt-get -y update --fix-missing \
     && apt-get -y upgrade \
-    && apt-get -y update \
-    && apt-get -y autoremove
-
+    && apt-get -y install software-properties-common \
+    && add-apt-repository ppa:deadsnakes/ppa \
+    && apt-get -y update
 # install dependencies
-RUN apt-get -y install \
+RUN apt-get -yqq install \
+    fontconfig \
     git \
-    python3.9 \
+    libglib2.0-0 \                                                                      
+    libsm6 \                                                                            
+    libxrender1 \                                                                       
+    libxext6 \
+    libfreetype6 \                                                                      
+    libgomp1 \
+    python3.12 \                                                                        
```

Comment:
EasyPQP requires Python 3.9, which has been included in the original dockerfile.